modules/light-clients/08-wasm/keeper/options_test.go

```diff
@@ -0,0 +1,151 @@
+				})
+				k = keeper.NewKeeperWithVM(
+					GetSimApp(suite.chainA).AppCodec(),
+					runtime.NewKVStoreService(GetSimApp(suite.chainA).GetKey(types.StoreKey)),
+					GetSimApp(suite.chainA).IBCKeeper.ClientKeeper,
+					GetSimApp(suite.chainA).WasmClientKeeper.GetAuthority(),
+					ibcwasm.GetVM(),
+					GetSimApp(suite.chainA).GRPCQueryRouter(),
+					querierOption,
+				)
+			},
+			func(k keeper.Keeper) {
+				plugins := types.GetQueryPlugins()
+
+				_, err := plugins.Custom(sdk.Context{}, nil)
+				suite.Require().ErrorContains(err, "custom querier error for TestNewKeeperWithOptions")
+
+				_, err = plugins.Stargate(sdk.Context{}, &wasmvmtypes.StargateQuery{})
+				suite.Require().ErrorIs(err, wasmvmtypes.UnsupportedRequest{Kind: "'' path is not allowed from the contract"})
+			},
+		},
+		{
+			"success: stargate querier",
+			func() {
+				querierOption := keeper.WithQueryPlugins(&types.QueryPlugins{
+					Stargate: MockStargateQuerier(),
+				})
+				k = keeper.NewKeeperWithVM(
+					GetSimApp(suite.chainA).AppCodec(),
+					runtime.NewKVStoreService(GetSimApp(suite.chainA).GetKey(types.StoreKey)),
+					GetSimApp(suite.chainA).IBCKeeper.ClientKeeper,
+					GetSimApp(suite.chainA).WasmClientKeeper.GetAuthority(),
+					ibcwasm.GetVM(),
+					GetSimApp(suite.chainA).GRPCQueryRouter(),
+					querierOption,
+				)
+			},
+			func(k keeper.Keeper) {
+				plugins := types.GetQueryPlugins()
+
+				_, err := plugins.Custom(sdk.Context{}, nil)
+				suite.Require().ErrorIs(err, wasmvmtypes.UnsupportedRequest{Kind: "Custom queries are not allowed"})
+
+				_, err = plugins.Stargate(sdk.Context{}, &wasmvmtypes.StargateQuery{})
+				suite.Require().ErrorContains(err, "stargate querier error for TestNewKeeperWithOptions")
+			},
+		},
+		{
+			"success: both queriers",
+			func() {
+				querierOption := keeper.WithQueryPlugins(&types.QueryPlugins{
+					Custom:   MockCustomQuerier(),
+					Stargate: MockStargateQuerier(),
+				})
+				k = keeper.NewKeeperWithVM(
+					GetSimApp(suite.chainA).AppCodec(),
+					runtime.NewKVStoreService(GetSimApp(suite.chainA).GetKey(types.StoreKey)),
+					GetSimApp(suite.chainA).IBCKeeper.ClientKeeper,
+					GetSimApp(suite.chainA).WasmClientKeeper.GetAuthority(),
+					ibcwasm.GetVM(),
+					GetSimApp(suite.chainA).GRPCQueryRouter(),
+					querierOption,
+				)
+			},
+			func(k keeper.Keeper) {
+				plugins := types.GetQueryPlugins()
+
+				_, err := plugins.Custom(sdk.Context{}, nil)
+				suite.Require().ErrorContains(err, "custom querier error for TestNewKeeperWithOptions")
+
+				_, err = plugins.Stargate(sdk.Context{}, &wasmvmtypes.StargateQuery{})
+				suite.Require().ErrorContains(err, "stargate querier error for TestNewKeeperWithOptions")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.SetupTest()
+
+		suite.Run(tc.name, func() {
+			tc.malleate()
+			tc.verifyFn(k)
+
+			// reset query plugins after each test
+			types.SetQueryPlugins(types.NewDefaultQueryPlugins())
```

Comment:
Maybe also do before call to `malleate`?

Reply:
done